plugins/wasm-rust/examples/http-dispatch-example/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2024 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use higress_wasm_rust::dispatcher::HttpDispatcher;
+use higress_wasm_rust::log::Log;
+use higress_wasm_rust::rule_matcher::{on_configure, SharedRuleMatcher};
+use higress_wasm_rust::{hostcalls, rule_matcher};
+use proxy_wasm::traits::{Context, HttpContext, RootContext};
+use proxy_wasm::types::{ContextType, HeaderAction, LogLevel};
+use serde::Deserialize;
+use std::ops::DerefMut;
+use std::rc::Rc;
+use std::time::Duration;
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_root_context(|_|Box::new(HttpDispatchExampleRoot::new()));
+}}
+
+struct HttpDispatchExampleRoot {
+    log: Rc<Log>,
+    rule_matcher: SharedRuleMatcher<HttpDispatchExampleConfig>,
+}
+
+struct HttpDispatchExample {
+    log: Rc<Log>,
+    rule_matcher: SharedRuleMatcher<HttpDispatchExampleConfig>,
+    http_dispatcher: HttpDispatcher,
+}
+
+#[derive(Default, Clone, Debug, Deserialize)]
+struct HttpDispatchExampleConfig {}
+
+impl HttpDispatchExampleRoot {
+    fn new() -> Self {
+        HttpDispatchExampleRoot {
+            log: Rc::new(Log::new("http-dispatch-example".to_string())),
+            rule_matcher: rule_matcher::new_shared(),
+        }
+    }
+}
+
+impl Context for HttpDispatchExampleRoot {}
+
+impl RootContext for HttpDispatchExampleRoot {
+    fn on_configure(&mut self, _plugin_configuration_size: usize) -> bool {
+        on_configure(
+            self,
+            _plugin_configuration_size,
+            self.rule_matcher.borrow_mut().deref_mut(),
+            &self.log,
+        )
+    }
+
+    fn create_http_context(&self, _context_id: u32) -> Option<Box<dyn HttpContext>> {
+        Some(Box::new(HttpDispatchExample {
+            log: self.log.clone(),
+            rule_matcher: self.rule_matcher.clone(),
+            http_dispatcher: Default::default(),
+        }))
+    }
+
+    fn get_type(&self) -> Option<ContextType> {
+        Some(ContextType::HttpContext)
+    }
+}
+
+impl Context for HttpDispatchExample {
+    fn on_http_call_response(
+        &mut self,
+        _token_id: u32,
+        _num_headers: usize,
+        _body_size: usize,
+        _num_trailers: usize,
+    ) {
+        self.http_dispatcher
+            .callback(_token_id, _num_headers, _body_size, _num_trailers)
+    }
+}
```

Comment:
This logic should be placed on the SDK side, so the user's code doesn't need to worry about this detail